aviatrix/resource_aviatrix_cloudn_edge_gateway.go

```diff
@@ -0,0 +1,300 @@
+	}
+
+	gateway := &goaviatrix.TransitVpc{
+		GwName: cloudnEdgeGateway.GatewayName,
+	}
+
+	_, prependAsPathOk := d.GetOk("prepend_as_path")
+	if _, ok := d.GetOk("local_as_number"); ok {
+		localASNumber := d.Get("local_as_number").(string)
+		err := client.SetLocalASNumber(gateway, localASNumber)
+		if err != nil {
+			return diag.Errorf("failed to create cloudn edge gateway: could not set local_as_number: %v", err)
+		}
+
+		if prependAsPathOk {
+			var prependASPath []string
+			for _, v := range d.Get("prepend_as_path").([]interface{}) {
+				prependASPath = append(prependASPath, v.(string))
+			}
+
+			err := client.SetPrependASPath(gateway, prependASPath)
+			if err != nil {
+				return diag.Errorf("failed to create cloudn edge gateway: could not set prepend_as_path: %v", err)
+			}
+		}
+	} else if prependAsPathOk {
+		return diag.Errorf("failed to create cloudn edge gateway: prepend_as_path must be empty when local_as_number has not been set")
+	}
+
+	return resourceAviatrixCloudnEdgeGatewayReadIfRequired(ctx, d, meta, &flag)
+}
+
+func resourceAviatrixCloudnEdgeGatewayReadIfRequired(ctx context.Context, d *schema.ResourceData, meta interface{}, flag *bool) diag.Diagnostics {
+	if !(*flag) {
+		*flag = true
+		return resourceAviatrixCloudnEdgeGatewayRead(ctx, d, meta)
+	}
+	return nil
+}
+
+func resourceAviatrixCloudnEdgeGatewayRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*goaviatrix.Client)
+
+	if d.Get("gw_name").(string) == "" {
+		id := d.Id()
+		log.Printf("[DEBUG] Looks like an import, no gw_name received. Import Id is %s", id)
+		d.Set("gw_name", id)
+		d.SetId(id)
+	}
+
+	cloudnEdgeGateway, err := client.GetCloudnEdgeGateway(ctx, d.Get("gw_name").(string))
+	if err != nil {
+		if err == goaviatrix.ErrNotFound {
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("failed to read CloudN Edge Gateway: %v", err)
+	}
+
+	d.Set("gw_name", cloudnEdgeGateway.GatewayName)
+	d.Set("over_private_network", cloudnEdgeGateway.OverPrivateNetwork)
+	d.Set("wan_interface_ip", cloudnEdgeGateway.WanInterfaceIp)
+	d.Set("wan_default_gateway", cloudnEdgeGateway.WanDefaultGateway)
+	d.Set("lan_interface_ip", cloudnEdgeGateway.LanInterfaceIp)
+	d.Set("management_interface_ip", cloudnEdgeGateway.ManagementInterfaceIp)
+	d.Set("default_gateway_ip", cloudnEdgeGateway.DefaultGatewayIp)
+	d.Set("dns_server", cloudnEdgeGateway.DnsServer)
+	d.Set("secondary_dns_server", cloudnEdgeGateway.SecondaryDnsServer)
+
+	if cloudnEdgeGateway.Dhcp {
+		d.Set("management_connection_type", "DHCP")
+	} else {
+		d.Set("management_connection_type", "Static")
+	}
+
+	gateway := &goaviatrix.TransitVpc{
+		GwName: d.Get("gw_name").(string),
+	}
+	transitGatewayAdvancedConfig, err := client.GetTransitGatewayAdvancedConfig(gateway)
+	if err != nil {
+		return diag.Errorf("failed to read cloudn edge gateway transit gateway advanced config: %v", err)
```

Comment:
Should be consistent on how the error messages capitalize the resource. Below you use CloudN Edge Gateway.